File: 2025-08-28/BookMyFood/src/app/service/food.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { IFood } from '../interfaces/food';
import { Observable } from 'rxjs';
import { BASE_URL } from '../util/appconstant';
import { AuthService } from './auth.service';


@Injectable({
  providedIn: 'root'
})
export class FoodService {
  constructor(private http: HttpClient, private authService: AuthService) {}

  getAllFoods(): Observable<IFood[]> {
    let token = localStorage.getItem('authToken');
    const headers = new HttpHeaders()
      .set('Content-Type', 'application/json')
      .set('Authorization', `Bearer ${token}`);

    let url: string = `${BASE_URL}/food`;
    return this.http.get<IFood[]>(url, { headers });
  }
}

File: 2025-08-28/foodms/src/main/java/com/wipro/foodms/service/FoodService.java
package com.wipro.foodms.service;

import com.wipro.foodms.entity.Food;

import java.util.List;

public interface FoodService {
    List<Food> getAllFoods();
    Food saveFood(Food food);
	Food findById(int id);
	void deleteById(int id);
}
